src/NHibernate/Properties/BasicPropertyAccessor.cs

```diff
@@ -132,14 +132,6 @@ internal static BasicSetter GetSetterOrNull(System.Type type, string propertyNam
 
 			BindingFlags bindingFlags = BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.DeclaredOnly;
 
-			if (type.IsValueType)
-			{
-				// the BindingFlags.IgnoreCase is important here because if type is a struct, the GetProperty method does
-				// not ignore case by default. If type is a class, it _does_ ignore case... we're better off explicitly
-				// stating that casing should be ignored so we get the same behavior for both structs and classes
-				bindingFlags = bindingFlags | BindingFlags.IgnoreCase;
-			}
```

Comment:
Yes but if we're going towards QueryAliasToObjectPropertySetter then I would suggest a separate PR for this then. I only did it here because it "was" used in AliasToBeanResultTransformer.